nim-compile.el

```diff
@@ -126,28 +105,26 @@ The config file would one of those: config.nims, PROJECT.nim.cfg, or nim.cfg."
    (delq nil `(,nim-compile-command ,@args ,@nim-compile-user-args ,file))
    " "))
 
-(define-derived-mode nim-compile-mode compilation-mode "nim-compile"
-  "major-mode for nim compilation buffer.")
-
 ;;;###autoload
 (defun nim-compile ()
   "Compile and execute the current buffer as a nim file.  All output is writton into the *compilation* buffer."
   (interactive)
   (when (derived-mode-p 'nim-mode)
-    (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
-    (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
-    (nim-compile--set-compile-command)
-    (funcall 'compile compile-command 'nim-compile-mode)))
+    ;; (add-hook 'compilation-filter-hook 'nim--colorize-compilation-buffer)
+    ;; (add-hook 'compilation-finish-functions 'nim--remove-colorize-hook)
+    ;; (nim-compile--set-compile-command)
+    ;; (funcall 'compile compile-command 'nim-compile-mode)))
+    (call-interactively 'compile)))
 
 (require 'ansi-color)
 (defun nim--colorize-compilation-buffer ()
-  "Colorize compilation buffer."
+  "Colorize compilation buffer.  Add to `compilation-filter-hook', when you want to interpret terminal colors in output buffer."
   (let ((inhibit-read-only t))
     (ansi-color-apply-on-region compilation-filter-start (point-max))))
 
 (defun nim--remove-colorize-hook (_buf _process-state)
-  "Remove ‘nim--colorize-compilation-buffer’."
-  (when (get-buffer "*nim-compile*")
+  "Remove ‘nim--colorize-compilation-buffer’.  Currently unused."
+  (when (get-buffer "*compilation*")
```

Comment:
well, for me this "other-cool-stull" just caused problems, and I removed it. For example the colorize hook and the removal. It just broke compatibility to "recompile", because when I hit recompile in a buffer that I just used for with nim-compile, the hook to colorize the buffer is gone and I have all the ugly symbols in the buffer. Also to have a new function that I need to call when I want to compile nim code is something I don't like. Because I can't use my normal keyboard binding I already bound to compile. I have to bind another key just to compile nim code. When you look at what nim-compile does now:

```elisp
(defun nim-compile ()
  "Compile and execute the current buffer as a nim file.  All output is writton into the *compilation* buffer."
  (interactive)
  (when (derived-mode-p 'nim-mode)
    (call-interactively 'compile)))
```
It is just an alias for compile, and I would like to keep it that way. Don't introduce logic and if statements. Just set compile-command in every nim-mode buffer once and locally, and that's it. There can be some logic setting that, but there should not be any nim-compile or anything like that. If people want their own compile function they can define it, but it should not be part of nim-mode. And because all nim-mode does now is setting compile-comand for the current active buffer, It can't conflict with other major modes.